lib/app/utils/terminology.rb

```diff
@@ -245,14 +247,35 @@ def self.validate_code(valueset_url: nil, code:, system: nil)
       end
 
       if system
-        validation_fn.call('code' => code, 'system' => system)
+        validation_fn.call(
+          'code' => uncoordinated_code(code: code, system: system, valueset_url: valueset_url),
+          'system' => system
+        )
       else
         @loaded_validators[valueset_url][:code_systems].any? do |possible_system|
-          validation_fn.call('code' => code, 'system' => possible_system)
+          validation_fn.call(
+            'code' => uncoordinated_code(code: code, system: possible_system, valueset_url: valueset_url),
+            'system' => possible_system
+          )
         end
       end
     end
 
+    def self.could_be_snomed_code?(valueset_url: nil, system: nil)
+      return false if system.present? && system != SNOMED_URL
+
+      system == SNOMED_URL || known_valuesets.dig(valueset_url, :code_systems)&.include?(SNOMED_URL)
+    end
+
+    # Return the base concept code for postcoordinated SNOMED codes
+    def self.uncoordinated_code(valueset_url: nil, code:, system: nil)
+      return code unless could_be_snomed_code?(valueset_url: valueset_url, system: system)
+
+      return code unless code&.include? '|'
+
+      code&.slice(0, code&.index('|'))
```

Comment:
You could also do `code&.split('|')&.first`, which might be more readable? Totally up to you though.